src/tasks/MobileBuildTasks/Android/Ndk/Ndk.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.Android.Build.Ndk
+{
+    public sealed class Ndk
+    {
+        private static string? ndkPath = "";
+        private static NdkVersion? ndkVersion;
+
+        public static string NdkPath
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(ndkPath))
+                {
+                    ndkPath = GetNdkPath(GetProbingPaths());
+                }
+
+                return ndkPath!;
+            }
+        }
+
+        public static NdkVersion NdkVersion
+        {
+            get
+            {
+#pragma warning disable IDE0074 // Use compound assignment
+                if (ndkVersion == null)
+                {
+                    ndkVersion = ReadVersion();
+                }
+#pragma warning restore IDE0074
+
+                return ndkVersion;
+            }
+        }
+
+        private static string? GetNdkPath(IEnumerable<string?> probingPaths)
+        {
+            string? ret = "";
+
+            foreach(string? path in probingPaths)
+            {
+                if (!string.IsNullOrEmpty(path) && Directory.Exists(path))
+                {
+                    ret = path;
+                    break;
+                }
+            }
+
+            return ret;
+        }
+
+        private static List<string?> GetProbingPaths()
+        {
+            List<string?> paths = new List<string?>();
+
+            string? ndkEnvPath = Environment.GetEnvironmentVariable("ANDROID_NDK_ROOT");
+
+            string[] fixedNdkPaths = (Utils.IsWindows()) ?
+                new string[]
+                {
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86), "Android", "android-sdk", "ndk-bundle"),
+                    Path.Combine(Environment.GetFolderPath (Environment.SpecialFolder.ProgramFilesX86), "Android", "android-sdk-windows", "ndk-bundle"),
+                    !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("ProgramW6432"))
+                        ? Path.Combine(Environment.GetEnvironmentVariable("ProgramW6432") ?? "", "Android", "android-sdk", "ndk-bundle")
+                        : Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "Android", "android-sdk", "ndk-bundle"),
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Android", "android-sdk", "ndk-bundle"),
+                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), "Android", "android-sdk", "ndk-bundle"),
+                    @"C:\android-sdk-windows"
```

Comment:
do we really want to support this path?

Reply:
That's what XA checks for as a fallback, so I just lifted it.